File: 알고리즘/이규준/20.06.08_printer/printer.py
def solution(priorities, location):
    q = []
    answer = 0

    for (num, priority) in enumerate(priorities):
        q.append((num, priority))

    while q:
        head = q.pop(0)

        if any(head[1] < i[1] for i in q):
            q.append(head)
        else:
            answer += 1

            if head[0] == location:
                break

    return answer

File: 알고리즘/김기강/20.06.08 printer/printer.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        
        Queue<qtem> queue = new ArrayDeque<>();
        for(int i = 0 ; i < priorities.length ; i++) {
            qtem tem  = new qtem(priorities[i]);
            if(i == location) {
                tem.ud = true;
            }
            queue.add(tem);
        }
        Arrays.sort(priorities);
        
        int where = priorities.length -1;
        
        while(!queue.isEmpty()) {
            qtem tem = queue.poll();
            if(tem.priority == priorities[where]) {
                if(tem.ud) {
                    return answer;
                }
                answer++;
                where--;
            } else {
                queue.add(tem);
            }   
        }
        
        return answer;
    }
}

class qtem {
    int priority;
    boolean ud;
    
    qtem(int priority) {
        this.priority = priority;
        this.ud = false;
    }
}